src/fn/watch-project.js

```diff
@@ -0,0 +1,206 @@
+    }
+};
+
+const waitForKillSignal = () => {
+    return new Promise((resolve) => {
+        process.on('SIGINT', async () => {
+            await cleanUp();
+            resolve();
+        });
+    });
+};
+
+const processAppForm = (fileName) => {
+    if (uploadForms.FORM_FILE_MATCHER(fileName)) {
+        eventQueue.enqueue(async () => {
+            await uploadAppForms([fileName.split('.')[0]]);
+            return fileName;
+        });
+        return true;
+    }
+    if (convertForms.FORM_FILE_MATCHER(fileName)) {
+        eventQueue.enqueue(async () => {
+            await convertAppForms([fileName.split('.')[0]]);
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const processAppFormMedia = (formMediaDir, fileName) => {
+    if (formMediaDir && formMediaDir.match(formMediaRegex)) {
+        eventQueue.enqueue(async () => {
+            await uploadAppForms([formMediaDir.split('-')[0]]);
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const processContactForm = (fileName) => {
+    if (convertForms.FORM_FILE_MATCHER(fileName)) {
+        eventQueue.enqueue(async () => {
+            await convertContactForm([fileName.split('.')[0]]);
+            return fileName;
+        });
+        return true;
+    }
+    if (uploadForms.FORM_FILE_MATCHER(fileName)) {
+        eventQueue.enqueue(async () => {
+            await uploadContactForms([fileName.split('.')[0]]);
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const processConfigFiles = (api, fileName) => {
+    if (fileName === 'app_settings.json') {
+        eventQueue.enqueue(async () => {
+            await uploadAppSettings(api);
+            return fileName;
+        });
+        return true;
+    }
+    if (fileName.match(/.+\.js$/) || fileName.match(/^[\w]+\.json$/)) {
+        eventQueue.enqueue(async () => {
+            await compileAppSettings();
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const watchProject = {
+    watch: async (api, blockFn, callback = {}, uploadStateOnStart = false) => {
+        if (uploadStateOnStart) {
+            await uploadInitialState(api);
+            info('Initial State Uploaded');
+        }
+
+        eventQueue = new Queue({ concurrent: 1, start: true });
+
+        fsEventsSubscription = await watcher.subscribe(environment.pathToProject, async (err, events) => {
+            if (err) {
+                error(err);
+                return;
+            }
+            for (const event of events) {
+                if (event.type !== 'update' && event.type !== 'create') {
+                    continue;
+                }
+                const changePath = event.path;
+                const parsedPath = path.parse(changePath);
+
+                if (parsedPath.dir.replace(environment.pathToProject, '').startsWith(`/${APP_FORMS_PATH}`)) {
```

Comment:
I think this will be more readable like: 
```suggestion
if (parsed.dir === path.join(environment.pathToProject, APP_FORMS_PATH))
```